src/utils/proxy.ts

```diff
@@ -151,12 +151,14 @@ export function getProxyRequestHeaders(event: H3Event) {
   return headers;
 }
 
-export function fetchWithEvent(
+export function fetchWithEvent<
+  T extends (req: RequestInfo | URL, opts?: any) => any = typeof fetch
```

Comment:
I think we might reserve first generic argument for fetch response type (same as $fetch, also reserving second and use this as third..)